### PR TITLE
feat: PyTorch training with GCS data

### PR DESCRIPTION
**Pull request by telpirion**

### CODE REVIEW

File: notebooks/official/training/pytorch_gcs_data_training.ipynb [hunk 1]
```diff
@@ -0,0 +1,1101 @@
+{
```
Comment (andrewferlitsch) [inline notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>Drop the first two steps: that's part of the setup</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1705/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (andrewferlitsch) [inline notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>Rewrite this in present tense: For this tutorial, a CSV version of the dataset available on <a href="https://www.kaggle.com/datasets/oddrationale/mnist-in-csv?select=mnist_train.csv" target="_blank">Kaggle</a> has been uploaded to a Cloud Storage bucket for your use.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1705/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (andrewferlitsch) [inline notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>The pip installs seem complicated. Consider using the template for install and restart</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1705/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (andrewferlitsch) [inline notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>use the notebook template</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1705/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (andrewferlitsch) [inline notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>use notebook template</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1705/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (andrewferlitsch) [inline notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>latest template does: f"gs://your-bucket-name-{PROJECT_ID}-unique"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1705/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

Comment (andrewferlitsch) [inline notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>URIs -&gt; Cloud Storage locations</p><p> </p><p>The (You can ...) should be in note form:L</p><p> </p><p>*Note:* You can ...</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1705/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='32'/>

Comment (andrewferlitsch) [inline notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>28 pixels wide by 28 pixels high -&gt; 28x28</p><p> </p><p>Anyways convention is height first then width</p><p> </p><p>I would drop "(a list of lists)".</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1705/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

Comment (andrewferlitsch) [inline notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>Add heding</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1705/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>

Comment (andrewferlitsch) [inline notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>add heading</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1705/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='41'/>

Comment (andrewferlitsch) [inline notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>Vertex -&gt; Vertex AI</p><p> </p><p>Although -&gt; While</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1705/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='44'/>

Comment (andrewferlitsch) [inline notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>Now you can write -&gt; Next, you write</p><p> </p><p>"to file" is redundant</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1705/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='49'/>

Comment (andrewferlitsch) [inline notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>"to file" is redundant</p><p> </p><p>you can now -&gt; you now</p><p> </p><p>Storage -&gt; Cloud Storage</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1705/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='52'/>

Comment (andrewferlitsch) [inline notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>delete the model and training job</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1705/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='55'/>

**Comment by andrewferlitsch** [comment]
File "/workspace/workspace/env/lib/python3.9/site-packages/nbclient/client.py", line 1005, in async_execute_cell
Step #5: Step #4:     raise DeadKernelError("Kernel died") from None
Step #5: Step #4: nbclient.exceptions.DeadKernelError: Kernel died

File: notebooks/official/training/pytorch_gcs_data_training.ipynb (hunk 1 above)
Comment (telpirion) [reply notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>That comes from the notebook template.</p><p> </p><p>https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb</p>

Comment (telpirion) [reply notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>Also from the template:</p><p>https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb</p>

Comment (telpirion) [reply notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>A managed model isn't created.</p>

Comment (telpirion) [reply notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>Which template--the notebook template? I think this is actually less complicated.</p><p> </p><p>Several libraries are required for this notebook. I think it's much cleaner and easier to read what libraries are added in this format.</p>

Comment (telpirion) [reply notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>Done.</p>

Comment (telpirion) [reply notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>(The future tense is what's to be avoided. See here: https://developers.google.com/style/tense)</p>

Comment (telpirion) [reply notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>s/"Cloud Storage Locations"/URIs: Ack.</p><p> </p><p>Note: done</p>

Comment (telpirion) [reply notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>Ack. This is intended for 100-level developers. It's intended to be more verbose and explanatory.</p>

Comment (telpirion) [reply notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>Ack. I don't think a heading is needed here.</p>

Comment (telpirion) [reply notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>Done.</p>

Comment (telpirion) [reply notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>s/Vertex/"Vertex AI": Done.</p><p>s/Although/While: Ack</p>

Comment (telpirion) [reply notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>"Although" is perfectly reasonable per word list: https://developers.google.com/style/word-list#letter-a</p>

Comment (telpirion) [reply notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>"To file": Ack (see previous).</p><p> </p><p>s/you can now/you now: Done.</p><p> </p><p>"Cloud Storage" is correct branding. See me if you have questions.</p>

Comment (telpirion) [reply notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>Done.</p>

File: notebooks/official/training/pytorch_gcs_data_training.ipynb [hunk 2]
```diff
@@ -0,0 +1,1173 @@
...
```
Comment (gericdong) [inline notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>Can you please check if the above install and uninstall are really necessary? Thanks.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1705/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (gericdong) [inline notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>BUCKET_PREFIX doesn't seem to be used anywhere else.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1705/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

Comment (telpirion) [reply notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>Changed. I had intended to use it later, for the output of the model.</p>

Comment (telpirion) [reply notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>Removed.</p>

File: notebooks/official/training/pytorch_gcs_data_training.ipynb (hunk 1 above)
Comment (gericdong) [reply notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>This is the notebook template: https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb</p><p> </p><p>Common libraries like pandas, numpy and matplotlib are pre-installed on Colab and Workbench environments. There is no need to install and update them.</p>

Comment (andrewferlitsch) [reply notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>i will accept this form, but for the future official notebooks must use the template version.</p>

Comment (andrewferlitsch) [reply notebooks/official/training/pytorch_gcs_data_training.ipynb:1]: <p>oops, correct</p>

**Review by andrewferlitsch — APPROVED** [review]

**Pull request merged**